### PR TITLE
zeroconf dns (bonjour) service publishing

### PR DESCRIPTION
**Pull request by samuraisam**
**What kind of change does this PR introduce?** Feature

**Did you add or update the `examples/`?** Yes

**Summary**
This creates a zeroconf dns service when the `--bonjour` flag is provided to the dev server command. it publishes an `http` service with the subtype of `webpack`. This is highly useful when debugging the live server across the network which can often be necessary when developing for network attached devices.

**Does this PR introduce a breaking change?**
No, but it does add a dependency on the `bonjour` package.

**Other information**
I had thought to add customization to the service type and subtypes but the defaults seem sufficient enough to serve the use case without bloating the interface.

### CODE REVIEW

**Comment by samuraisam** [comment]
I'm actually not sure why travis would fail here.

File: bin/webpack-dev-server.js [hunk 1]
```diff
@@ -401,8 +409,10 @@ function startDevServer(wpOpt, options) {
 			});
 		});
 	} else {
+		console.log("shits");
```
Comment (shellscape) [inline bin/webpack-dev-server.js]: well I'm not sure we want to be telling users about this 😄

**Comment by shellscape** [comment]
OK, you need to add the option to https://github.com/webpack/webpack-dev-server/blob/master/test/Validation.test.js#L51. See https://github.com/webpack/webpack-dev-server/commit/ab889c31a584e193814b2e952410307d071ede61#diff-46ade4578831d079715c6efd9c726d48 for an example

**Comment by shellscape** [comment]
I think you missed the test file in that last commit :)

**Comment by samuraisam** [comment]
Wow this has to be my personal record for worst managed PR ever. Thanks for being understanding!

**Comment by shellscape** [comment]
@samuraisam no worries :) PR is solid, but the CLA needs signing (https://github.com/webpack/webpack-dev-server/pull/930#issuecomment-306384164) before we can merge it. Please give that a look.

**Comment by samuraisam** [comment]
It looks like I have signed it (the CLA check passes)

**Pull request merged**